service/history/handler.go

```diff
@@ -924,6 +924,30 @@ func (h *Handler) ReplicateEvents(ctx context.Context, replicateRequest *hist.Re
 	return nil
 }
 
+// SyncShardStatus is called by processor to sync history shrad information from another cluster
+func (h *Handler) SyncShardStatus(ctx context.Context, syncShardStatusRequest *hist.SyncShardStatusRequest) error {
+	h.startWG.Wait()
+
+	h.metricsClient.IncCounter(metrics.HistorySyncShardStatusScope, metrics.CadenceRequests)
+	sw := h.metricsClient.StartTimer(metrics.HistorySyncShardStatusScope, metrics.CadenceLatency)
+	defer sw.Stop()
+
+	// shard ID is already provided in the request
+	engine, err := h.controller.getEngineForShard(int(syncShardStatusRequest.GetShardId()))
```

Comment:
Can you validate if the request has shardID and return BadRequest if it does not.

Reply:
sure